lib/broadway_kafka/brod_client.ex

```diff
@@ -125,6 +125,41 @@ defmodule BroadwayKafka.BrodClient do
     :ok
   end
 
+  @impl true
+  def disconnect(nil, client_id) do
+    :ok = :brod.stop_client(client_id)
+    :ok
+  end
+
+  def disconnect(group_coordinator, client_id) do
+    ref = Process.monitor(group_coordinator)
+    Process.exit(group_coordinator, :kill)
+
+    receive do
+      {:DOWN, ^ref, _, _, _} -> :ok
+    end
+    :ok = :brod.stop_client(client_id)
+    :ok
+  end
+
+
+  @impl true
+  def disconnect(nil, client_id) do
+    :ok = :brod.stop_client(client_id)
```

Comment:
Disconnect is duplicated now. Although you should make it so disconnect only stops the client and the producer can call both stop_group_coordinator and disconnect if necessary.

Reply:
Done, thank you!